src/Components/PluginZip.php

```diff
@@ -31,9 +31,18 @@ public function zip(string $directory, ?string $branch = null): string
         // Extract the git repository there
         $this->exec(sprintf('git archive %s | tar -x -C %s', escapeshellarg($branch), escapeshellarg($pluginName)));
 
-        // Install composer dependencies
-        if ($this->needComposerToRun($directory)) {
-            $this->exec('composer install --no-dev -n -o -d ' . escapeshellarg($pluginName));
+        $composerJson = $directory . '/' . $pluginName . '/composer.json';
+        $composerJsonBackup = $composerJson . '.bak';
+
+        if (file_exists($composerJson)) {
+            copy($composerJson, $composerJsonBackup);
+            $this->filterShowpareDependencies($composerJson);
+            // Install composer dependencies
+            if ($this->needComposerToRun($composerJson)) {
+                $this->exec('composer install --no-dev -n -o -d ' . escapeshellarg($pluginName));
+            }
+
+            rename($composerJsonBackup, $composerJson);
```

Comment:
Does this rename work? Cause the $composerJson is not deleted? 🤔

Reply:
I just added the "rename" literally a minute ago. You probably still have "copy" checked out.